src/modules/Packages.rb

```diff
@@ -2623,6 +2625,35 @@ def report_missing_pattern(pattern_name)
       end
     end
 
+    # Search for providers for a list of tags
+    #
+    # The use case of this method is to convert and array
+    # of tags into an array of packages.
+    #
+    # @param tags [Array<String>] List of tags (ie. package names) to search for.
+    # @return     [Array<String>] List contaning a package for each tag.
+    # @see find_provider
+    def find_providers(tags)
+      tags.map { |tag| find_provider(tag) }
+    end
+
+    # Search a provider for a tag
+    #
+    # If a provider is not found, an error will be logged.
+    #
+    # @param tag [String] Tag to search a package for.
+    # @return    [String] Name of the package which provides that tag.
+    # @see find_providers
+    def find_provider(tag)
+      provided_by = Pkg.PkgQueryProvides(tag).find { |provide| provide[1] != :NONE }
+      if provided_by.nil?
+        log.error "Provider not found for '#{tag}'"
+        tag
+      else
+        provided_by.first
```

Comment:
You should at least log the ignored packages if there are more providers of `tag`. YaST might choose a "random" package if more providers are present. This would help with debugging later...